### ISSUE TITLE
Bug in keepalive logic causes i/o timeout error

### ISSUE THREAD

**Issue by spacepille** [vcabbage/amqp#164]
Hi there,
I found a bug in master branch in conn.go:595 in keepalive case. When connectTimout is set, writeFrame func uses c.net.SetWriteDeadline(). The keepalive case send directly to net without recognizing a write deadline. It causes an i/o timeout error. Please see the fix below. I also send you a pull request on this issue.

```
// keepalive timer
case <-keepalive:
  if c.connectTimeout != 0 {
    _ = c.net.SetWriteDeadline(time.Now().Add(c.connectTimeout))
  }
  _, err = c.net.Write(keepaliveFrame)
```

**Comment by vcabbage** [comment]
Thanks for reporting this!

The `connectTimeout` isn't supposed to apply after the initial connection is made. I think the approach to resolving this is to have `conn.connWriter()` clear the write timeout and set `c.connectTimeout = 0`.

Let me know what you think and whether you'd like to update your existing PR.

**Comment by vcabbage** [comment]
Fixed in #163. Tagged as v0.11.2.
